### PR TITLE
fix(ios): Crash prevention for (some) malformed language data from API

### PR DESCRIPTION
**Pull request by jahorton**
This should prevent another :fire: from occurring like the one triggered by the issues addressed by https://github.com/keymanapp/api.keyman.com/pull/20.

The app will now handle the failure state, rather than crashing when attempting to create a `Dictionary` with duplicate ids.  Unfortunately, Swift does not allow `try`-`catch`ing that particular error, so the language `Dictionary`'s initialization had to be rewritten accordingly.

Should the scenario ever arise again, we'll be able to see the following text in the app's log messages:

```
2019-11-22 10:54:29.435 [Error] [APIKeyboardRepository.swift:100] apiCompletionHandler(data:response:error:fetchCompletionHandler:) > Corrupt data detected in API returns - duplicate language codes: [en-us, fr-ca]
```

(Of course, the language codes in the message above are just examples.)

### CODE REVIEW

File: ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APIKeyboardRepository.swift [hunk 1]
```diff
@@ -75,7 +76,32 @@ public class APIKeyboardRepository: KeyboardRepository {
...
+      }
+
+      dictionary.updateValue(language, forKey: language.id)
+    }
+
+    guard !duplicateDetected else {
+      // This indicates an issue in our keyboard API infrastructure.
+      // Don't let it crash the app, but definitely make note of it
+      // as something to be fixed.
+      log.error("Corrupt data detected in API returns: duplicate language codes")
```
Comment (darcywong00) [inline ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APIKeyboardRepository.swift]: To help with future troubleshooting, can this include the duplicated language code?

Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APIKeyboardRepository.swift]: Might have to be "code**s**"... it should be possible to track an array for this.  Assuming there's only one may be a bit naive.

Comment (darcywong00) [reply ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APIKeyboardRepository.swift]: Good point. If  the API goes really pear-shaped, maybe just a handful of codes will suffice to see what's going on.

File: ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APIKeyboardRepository.swift [hunk 2]
```diff
@@ -75,7 +76,32 @@ public class APIKeyboardRepository: KeyboardRepository {
     }
 
     options = result.options
-    languages = Dictionary(uniqueKeysWithValues: result.languages.map { ($0.id, $0) })
+
+    // Detect duplicate ids - this scenario indicates an infrastructure problem.
+    var dictionary = Dictionary<String, Language>()
```
Comment (mcdurdin) [inline ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APIKeyboardRepository.swift]: Wow. I think I preferred the earlier code for readability! Could you refactor this into a function?

**Pull request merged**